src/symbolize/source.rs

```diff
@@ -108,25 +108,60 @@ impl Debug for Process {
     }
 }
 
-impl From<Process> for Source {
+impl From<Process> for Source<'static> {
     fn from(process: Process) -> Self {
         Source::Process(process)
     }
 }
 
 
-/// A gsym file.
 #[derive(Clone, Debug)]
-pub struct Gsym {
-    /// The path to the gsym file.
+pub enum Gsym<'dat> {
+    /// "Raw" Gsym data.
+    Data(GsymData<'dat>),
+    /// A Gsym file.
+    File(GsymFile),
+}
```

Comment:
We could think about flattening that into `Source` directly. Though, long term I'd rather make `Source` a completely internal thing, to be honest.